addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/hw/HwSerialBridgeHandler.java

```diff
@@ -0,0 +1,275 @@
+            serialPort = portIdentifier.open(this.getClass().getName(), 2000);
+
+            logger.debug("Connection established using {}.  Configuring IO parameters. ", serialPortName);
+
+            int db = SerialPort.DATABITS_8, sb = SerialPort.STOPBITS_1, p = SerialPort.PARITY_NONE;
+            serialPort.setSerialPortParams(baudRate, db, sb, p);
+            serialPort.enableReceiveThreshold(1);
+            serialPort.disableReceiveTimeout();
+            serialOutput = new OutputStreamWriter(serialPort.getOutputStream(), "US-ASCII");
+            serialInput = new BufferedReader(new InputStreamReader(serialPort.getInputStream(), "US-ASCII"));
+
+            serialPort.addEventListener(this);
+            serialPort.notifyOnDataAvailable(true);
+
+            logger.debug("Sending monitoring commands.");
+            sendCommand("PROMPTOFF");
+            sendCommand("KBMOFF");
+            sendCommand("KLMOFF");
+            sendCommand("GSMOFF");
+            sendCommand("DLMON"); // Turn on dimmer monitoring
+
+            logger.info("Setting status to ONLINE");
+            updateStatus(ThingStatus.ONLINE);
+
+            if (updateTime) {
+                startUpdateProcessorTimeJob();
+            }
+
+        } catch (NoSuchPortException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Invalid port: " + serialPortName);
+        } catch (PortInUseException portInUseException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Port in use: " + serialPortName);
+        } catch (UnsupportedCommOperationException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Communication error");
+        } catch (TooManyListenersException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error configuring serial port.");
+        }
+    }
+
+    private void startUpdateProcessorTimeJob() {
+        if (updateTimeJob != null) {
+            logger.debug("Canceling old scheduled job");
+            updateTimeJob.cancel(false);
+            updateTimeJob = null;
+        }
+
+        logger.debug("Scheduling time update job for 2:00 AM");
+
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime twoAm = now.withHour(2).withMinute(0).withSecond(0);
+        if (now.compareTo(twoAm) > 0) {
+            twoAm = twoAm.plusDays(1);
+        }
+
+        Duration duration = Duration.between(now, twoAm);
+        long initialDelay = duration.getSeconds();
+        long delay = Duration.ofDays(1).getSeconds();
+        TimeUnit unit = TimeUnit.SECONDS;
+
+        logger.debug("Update Time job will run in {} and every 1 day thereafter.", duration);
+        updateTimeJob = this.scheduler.scheduleWithFixedDelay(() -> updateProcessorTime(), initialDelay, delay, unit);
```

Comment:
Why is it so important to update at two AM, the code because 10x more complex because of that otherwise you could just use (replacing line 141 - 156):

```
updateTimeJob = scheduler.scheduleWithFixedDelay(() -> updateProcessorTime(), 0, 1, TimeUnit.DAYS);
```